library.go

```diff
@@ -1827,6 +1826,49 @@ func string_split(fnname string, recv_ Value, args Tuple, kwargs []Tuple) (Value
 	return NewList(list), nil
 }
 
+func reverse(s string, size int) string {
+	rev := make([]rune, 0, size)
+
+	for _, point := range s {
+		rev = append(rev, point)
+	}
+
+	for i := 0; i < size/2; i++ {
+		rev[i], rev[size-1-i] = rev[size-1-i], rev[i]
+	}
+
+	return string(rev)
+}
+
+func rsplit(s string, max int) []string {
+	size := utf8.RuneCountInString(s)
+	splitCount := 0   // Our current number of splits made.
+	currIndex := size // A backward working index.
+	lastStop := size  // After we make a split, advance this back.
+	parts := make([]string, 0, max)
+
+	reversed := reverse(s, size)
+
+	for _, p := range reversed {
```

Comment:
This loop does UTF-8 decoding a second time.  It would be more efficient (and probably fewer lines of code) to work backwards, identifying the start of each UTF-8 encoded code point, in a single pass.

Reply:
I am just going to iterate through the slice as noted above.